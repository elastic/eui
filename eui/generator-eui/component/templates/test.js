import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../test';

import { <%= componentName %> } from './<%= fileName %>';

describe('<%= componentName %>', () => {
  test('is rendered', () => {
    const component = render(
      <<%= componentName %> {...requiredProps} />
    );

    expect(component)
      .toMatchSnapshot();
  });
});
