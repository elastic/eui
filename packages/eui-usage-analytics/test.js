/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

const Codeowners = require('codeowners');
const temp = new Codeowners("../../../kibana");

const { scan } = require('./scan');

const runScan = async () => {
  const scanResult = await scan();
  console.log(scanResult);
};

runScan();
