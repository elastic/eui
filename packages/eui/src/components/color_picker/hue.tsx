/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, {
  ChangeEvent,
  InputHTMLAttributes,
  FunctionComponent,
} from 'react';
import classNames from 'classnames';

import { useEuiMemoizedStyles } from '../../services';
import { CommonProps } from '../common';
import { EuiScreenReaderOnly } from '../accessibility';
import { EuiI18n, useEuiI18n } from '../i18n';

import { euiHueStyles } from './hue.styles';
import { EuiToolTip } from '../tool_tip';

const HUE_RANGE = 359;

export type EuiHueProps = Omit<
  InputHTMLAttributes<HTMLInputElement>,
  'onChange'
> &
  CommonProps & {
    hex?: string;
    hue?: string | number;
    onChange: (hue: number) => void;
  };

export const EuiHue: FunctionComponent<EuiHueProps> = ({
  className,
  hex,
  hue = 1,
  id,
  onChange,
  ...rest
}) => {
  const classes = classNames('euiHue', className);
  const styles = useEuiMemoizedStyles(euiHueStyles);

  const [ariaValueText, ariaRoleDescription] = useEuiI18n(
    ['euiHue.ariaValueText', 'euiHue.ariaRoleDescription'],
    ['Hue', 'Hue slider']
  );

  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    onChange(Number(e.target.value));
  };

  const hueValue = typeof hue === 'string' ? parseInt(hue) : hue;
  // align the tooltip contextually closer to the thumb
  const tooltipPosition =
    hueValue < Math.floor(HUE_RANGE / 2) ? 'left' : 'right';

  return (
    <div css={styles.euiHue} className={classes}>
      <EuiScreenReaderOnly>
        <label htmlFor={`${id}-hue`}>
          <EuiI18n
            token="euiHue.label"
            default="Select the HSV color mode 'hue' value"
          />
        </label>
      </EuiScreenReaderOnly>
      {/* we can only wrap the entire input because the input slider thumb is not a standalone element */}
      <EuiToolTip
        content={hex}
        anchorProps={{
          css: styles.euiHue__tooltip,
        }}
        disableScreenReaderOutput
        position={tooltipPosition}
      >
        <input
          id={`${id}-hue`}
          min={0}
          max={HUE_RANGE}
          step={1}
          type="range"
          css={styles.euiHue__range}
          className="euiHue__range"
          value={hue}
          onChange={handleChange}
          aria-roledescription={ariaRoleDescription}
          aria-valuetext={`${ariaValueText} ${hue}Â°`}
          {...rest}
        />
      </EuiToolTip>
    </div>
  );
};
