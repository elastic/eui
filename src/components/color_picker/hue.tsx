/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, {
  ChangeEvent,
  InputHTMLAttributes,
  FunctionComponent,
} from 'react';
import classNames from 'classnames';
import { CommonProps } from '../common';

import { EuiScreenReaderOnly } from '../accessibility';
import { EuiI18n } from '../i18n';

const HUE_RANGE = 359;

export type EuiHueProps = Omit<
  InputHTMLAttributes<HTMLInputElement>,
  'onChange'
> &
  CommonProps & {
    hex?: string;
    hue?: string | number;
    onChange: (hue: number) => void;
  };

export const EuiHue: FunctionComponent<EuiHueProps> = ({
  className,
  hex,
  hue = 1,
  id,
  onChange,
  ...rest
}) => {
  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    onChange(Number(e.target.value));
  };
  const classes = classNames('euiHue', className);
  return (
    <React.Fragment>
      <EuiScreenReaderOnly>
        <label htmlFor={`${id}-hue`}>
          <EuiI18n
            token="euiHue.label"
            default="Select the HSV color mode 'hue' value"
          />
        </label>
      </EuiScreenReaderOnly>
      <EuiScreenReaderOnly>
        <p aria-live="polite">{hex}</p>
      </EuiScreenReaderOnly>
      <div className={classes}>
        <input
          id={`${id}-hue`}
          min={0}
          max={HUE_RANGE}
          step={1}
          type="range"
          className="euiHue__range"
          value={hue}
          onChange={handleChange}
          {...rest}
        />
      </div>
    </React.Fragment>
  );
};
