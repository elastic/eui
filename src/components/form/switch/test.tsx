import React from 'react';

import { EuiSwitch } from './switch';

export const Test = ({ checked }: { checked: boolean }) => (
  <EuiSwitch
    checked={checked}
    label="Test"
    onChange={e => console.log(e.target.checked)}
  />
);
