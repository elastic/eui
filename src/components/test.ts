export function add(x: number, y: number) {
  return x + y;
}

add(5, 4);
