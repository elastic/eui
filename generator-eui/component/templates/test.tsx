/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { render } from '../../test/rtl';
import { requiredProps } from '../../test/required_props';

import { <%= componentName %> } from './<%= fileName %>';

describe('<%= componentName %>', () => {
  it('renders', () => {
    const { container } = render(
      <<%= componentName %> {...requiredProps} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
