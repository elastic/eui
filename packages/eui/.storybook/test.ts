/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { queries, within, waitFor, fireEvent, expect } from '@storybook/test';
import * as dataTestSubjQueries from '../src/test/rtl/data_test_subj_queries';

/**
 * Custom Storybook within util with EUI query helpers
 * + additional chained async/waitFor component utils
 *
 * @see https://storybook.js.org/docs/writing-stories/play-function#writing-stories-with-the-play-function
 * @see https://testing-library.com/docs/dom-testing-library/api-within/
 */
const customWithin = (canvasElement: HTMLElement) => {
  const canvas = within<typeof queries & typeof dataTestSubjQueries>(
    canvasElement,
    { ...queries, ...dataTestSubjQueries }
  );

  return {
    ...canvas,

    /**
     * 1. Loki doesn't like userEvent, only fireEvent
     * 2. Storybook fires fireEvents too early (esp. on page load), so we add a waitFor
     */
    waitForAndClick: async (testSubject: string) => {
      await waitFor(() =>
        expect(canvas.getByTestSubject(testSubject)).toBeInTheDocument()
      );
      await fireEvent.click(canvas.getByTestSubject(testSubject));
    },

    waitForEuiPopoverVisible: async () =>
      await waitFor(() =>
        expect(canvasElement.querySelector('[data-popover-open]')).toBeVisible()
      ),
    waitForEuiPopoverHidden: async () =>
      await waitFor(() =>
        expect(
          canvasElement.querySelector('[data-popover-panel]')
        ).not.toBeInTheDocument()
      ),
  };
};

export { customWithin as within };
