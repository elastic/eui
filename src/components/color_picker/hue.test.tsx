/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../test/required_props';

import { EuiHue } from './hue';

const onChange = () => {
  /* empty */
};

describe('EuiHue', () => {
  test('is rendered', () => {
    const component = render(<EuiHue onChange={onChange} {...requiredProps} />);

    expect(component).toMatchSnapshot();
  });

  test('accepts a hue value', () => {
    const component = render(
      <EuiHue hue={180} onChange={onChange} {...requiredProps} />
    );

    expect(component).toMatchSnapshot();
  });

  test('accepts a hex value', () => {
    const component = render(
      <EuiHue hue={180} hex="#00FFFF" onChange={onChange} {...requiredProps} />
    );

    expect(component).toMatchSnapshot();
  });
});
