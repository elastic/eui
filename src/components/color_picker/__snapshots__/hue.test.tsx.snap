// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiHue accepts a hex value 1`] = `
Array [
  <label
    class="euiScreenReaderOnly"
    for="undefined-hue"
  >
    Select the HSV color mode 'hue' value
  </label>,
  <p
    aria-live="polite"
    class="euiScreenReaderOnly"
  >
    #00FFFF
  </p>,
  <div
    class="euiHue testClass1 testClass2"
  >
    <input
      aria-label="aria-label"
      class="euiHue__range"
      data-test-subj="test subject string"
      id="undefined-hue"
      max="359"
      min="0"
      step="1"
      type="range"
      value="180"
    />
  </div>,
]
`;

exports[`EuiHue accepts a hue value 1`] = `
Array [
  <label
    class="euiScreenReaderOnly"
    for="undefined-hue"
  >
    Select the HSV color mode 'hue' value
  </label>,
  <p
    aria-live="polite"
    class="euiScreenReaderOnly"
  />,
  <div
    class="euiHue testClass1 testClass2"
  >
    <input
      aria-label="aria-label"
      class="euiHue__range"
      data-test-subj="test subject string"
      id="undefined-hue"
      max="359"
      min="0"
      step="1"
      type="range"
      value="180"
    />
  </div>,
]
`;

exports[`EuiHue is rendered 1`] = `
Array [
  <label
    class="euiScreenReaderOnly"
    for="undefined-hue"
  >
    Select the HSV color mode 'hue' value
  </label>,
  <p
    aria-live="polite"
    class="euiScreenReaderOnly"
  />,
  <div
    class="euiHue testClass1 testClass2"
  >
    <input
      aria-label="aria-label"
      class="euiHue__range"
      data-test-subj="test subject string"
      id="undefined-hue"
      max="359"
      min="0"
      step="1"
      type="range"
      value="1"
    />
  </div>,
]
`;
